### ISSUE TITLE
 VARCHAR/CHAR.. column NOT NULLABLE but DEFAULT NULL may should show Default NULL(not blank) in table structure view

### ISSUE THREAD

**Issue by mrrao** [Sequel-Ace/Sequel-Ace#1419]
- Sequel Ace Version (including build number): Until Version 3.5.1 Build 20031
- MySQL Version: Mysql 8.0.28
```
mysql> SHOW CREATE TABLE `test`.`test`\G
*************************** 1. row ***************************
       Table: test
Create Table: CREATE TABLE `test` (
  `id` int unsigned NOT NULL,
  `name` varchar(8) CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci NOT NULL,
  `type` char(8) NOT NULL DEFAULT ''
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
1 row in set (0.00 sec)

mysql> SELECT `COLUMN_NAME`, `DATA_TYPE`, `COLUMN_TYPE`, `IS_NULLABLE`, `COLUMN_DEFAULT`
    ->     FROM `information_schema`.`COLUMNS`
    ->    WHERE `TABLE_SCHEMA` = "test"
    ->      AND `TABLE_NAME` = "test"
    -> ORDER BY `ORDINAL_POSITION`;
+-------------+-----------+--------------+-------------+----------------+
| COLUMN_NAME | DATA_TYPE | COLUMN_TYPE  | IS_NULLABLE | COLUMN_DEFAULT |
+-------------+-----------+--------------+-------------+----------------+
| id          | int       | int unsigned | NO          | NULL           |
| name        | varchar   | varchar(8)   | NO          | NULL           |
| type        | char      | char(8)      | NO          |                |
+-------------+-----------+--------------+-------------+----------------+
3 rows in set (0.00 sec)
```
`name` DEFAULT VALUE is different from `type`, but in table structure view they are same:
![img](https://user-images.githubusercontent.com/3365281/158157003-fc648c73-c0a6-4df2-9527-e37a7cb6b426.png)

In SQL, DEFAULT NULL is different from "", even column NOT NULLABLE:
```
mysql> INSERT INTO `test`.`test` (`id`, `name`, `type`) value (1, DEFAULT, "TYPE");
ERROR 1364 (HY000): Field 'name' doesn't have a default value
mysql> INSERT INTO `test`.`test` (`id`, `name`, `type`) value (1, "NAME", DEFAULT);
Query OK, 1 row affected (0.00 sec)

mysql> SELECT * FROM `test`.`test`;
+----+------+------+
| id | name | type |
+----+------+------+
|  1 | NAME |      |
+----+------+------+
1 row in set (0.00 sec)
```
DEFAULT INSERT can be used in dynamic SQL(if column null use DEFAULT). INSERT NULL `name` cause ERROR, but INSERT "" `type` success silently(cause dirty data).
No "string not empty" constraints before MYSQL 8.0.16, I use "DEFAULT NULL NOT NULL" to prevent forget to set the required column.
Hope to see Default NULL clearly in structure ui.

Suggest:
![WechatIMG7](https://user-images.githubusercontent.com/3365281/158157821-69b799a2-c482-49a0-b3da-e9f447cf82c8.png)

**Comment by Jason-Morcos** [comment]
<img width="260" alt="Screen Shot 2022-03-14 at 17 26 52" src="https://user-images.githubusercontent.com/10710367/158321124-7ef72493-9b65-41a2-9693-7f27a987cf61.png">
I get an error trying to create a column that's not null default null in MySQL 5.7

When I dug into this, it looks like the UI isn't showing default blank string, it's showing that there's _no default_ (i.e. blank). Functionally, NOT NULL DEFAULT NULL is _having no default value at all_, right? Because you'll error on insert both if you don't specify a value or if you don't specify a value and if falls back on NULL. I don't think it makes sense for most users to change the behavior in this regard.

@gduh maybe we could somehow tweak the UI to wrap all empty strings in quotes like it does when actually generating the alter statement (thanks to that great work you did!)

**Comment by mrrao** [comment]
DEFAULT NULL means no default value, but different from "" if column is string. Distinguish them more clearly in UI may be better.

**Comment by mrrao** [comment]
"NOT NULL DEFAULT NULL is having no default value at all, right?" -> right
"Because you'll error on insert both if you don't specify a value or if you don't specify a value and if falls back on NULL." ->
```
mysql> CREATE TABLE `test`.`test1` (
    ->   `id` int NOT NULL,
    ->   `name` varchar(8) NOT NULL
    -> );
Query OK, 0 rows affected (0.00 sec)

mysql> DESCRIBE `test`.`test1`;
+-------+------------+------+-----+---------+-------+
| Field | Type       | Null | Key | Default | Extra |
+-------+------------+------+-----+---------+-------+
| id    | int        | NO   |     | NULL    |       |
| name  | varchar(8) | NO   |     | NULL    |       |
+-------+------------+------+-----+---------+-------+
2 rows in set (0.00 sec)

mysql> INSERT INTO `test`.`test1`(`id`) VALUE(1);
ERROR 1364 (HY000): Field 'name' doesn't have a default value
mysql> CREATE TABLE `test`.`test2` (
    ->   `id` int NOT NULL,
    ->   `name` varchar(8) NOT NULL DEFAULT ""
    -> );
Query OK, 0 rows affected (0.01 sec)

mysql> DESCRIBE `test`.`test2`;
+-------+------------+------+-----+---------+-------+
| Field | Type       | Null | Key | Default | Extra |
+-------+------------+------+-----+---------+-------+
| id    | int        | NO   |     | NULL    |       |
| name  | varchar(8) | NO   |     |         |       |
+-------+------------+------+-----+---------+-------+
2 rows in set (0.00 sec)

mysql> INSERT INTO `test`.`test2`(`id`) VALUE(1);
Query OK, 1 row affected (0.01 sec)
```
NOT NULL DEFAULT NULL will error, that's I want, because the bug will be found earlier;
NOT NULL DEFAULT "" will success, that's not I want, because the inserted data is invalid, but no error!

<strong>Can't clearly distinguish DEFAULT NULL or DEFAULT "" in UI, that's the only problem.</strong>

**Comment by mrrao** [comment]
"I get an error trying to create a column that's not null default null in MySQL 5.7" ->
```
ALTER TABLE `report_configurations` ALTER COLUMN `test` DROP DEFAULT
```
Because column NOT NULL, can't SET DEFAULT NULL(Invalid default value), but you can drop it.

**Comment by mrrao** [comment]
In fact, DEFAULT NULL is more “supernatural”:
```sql
mysql> CREATE TABLE `test`.`test3` (
    ->   `id` int NULL DEFAULT NULL
    -> );
Query OK, 0 rows affected (0.01 sec)

mysql> DESCRIBE `test`.`test3`;
+-------+------+------+-----+---------+-------+
| Field | Type | Null | Key | Default | Extra |
+-------+------+------+-----+---------+-------+
| id    | int  | YES  |     | NULL    |       |
+-------+------+------+-----+---------+-------+
1 row in set (0.00 sec)

mysql> INSERT INTO `test`.`test3` value();
Query OK, 1 row affected (0.00 sec)

mysql> SELECT * FROM `test`.`test3`;
+------+
| id   |
+------+
| NULL |
+------+
1 row in set (0.00 sec)

mysql> ALTER TABLE `test`.`test3` ALTER COLUMN `id` DROP DEFAULT;
Query OK, 0 rows affected (0.01 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql> DESCRIBE `test`.`test3`;
+-------+------+------+-----+---------+-------+
| Field | Type | Null | Key | Default | Extra |
+-------+------+------+-----+---------+-------+
| id    | int  | YES  |     | NULL    |       |
+-------+------+------+-----+---------+-------+
1 row in set (0.00 sec)

mysql> INSERT INTO `test`.`test3` value();
ERROR 1364 (HY000): Field 'id' doesn't have a default value
```
Tested both on Mysql 5.7 and 8.0.

"Having default value NULL(DEFAULT NULL)" is different from "Having no default value(DROP)" in action.

But they are all "DEFAULT NULL" in DESCRIBE command, even in information_schema.COLUMNS, I can't find any difference .

**Comment by gduh** [comment]
In the end, the difference between `SHOW CREATE TABLE xxxx` and `SELECT xxxxx FROM information_schema.COLUMNS` is confusing, isn't it ?

While testing, I found an interesting thing in SACE in the `QUERY` tab. Using the `SELECT`
<img width="617" alt="image" src="https://user-images.githubusercontent.com/15440947/158551890-ff7b149f-f84b-4270-91cb-3e0fe73a7a4f.png">
I didn't go to check in the code, but it looks like SACE displays the `NULL` in either gray or white depending on how it's represented in the response.

@Jason-Morcos 
Anyway, I think the display in the 'Structure' tab needs to be clarified : 
- Adding '' in case of `default ''` is set.
- Globally, for any `default 'xxxx'`, add the '' if required: 
    - `DEFAULT 'SACE'`  -->  `'SACE'`   (With '')
    - `DEFAULT 123`  -->  `123`    (Without '')
- Adding NULL (grey?) if NULL is not allowed and "DEFAULT VALUE" is not set?

About `DROP DEFAULT, well, how many people do that, really ?

**Comment by mrrao** [comment]
`DEFAULT VALUE`: 
    * `not exists`
    * `NULL` 
    * `''`(empty string)
    * `' '`(blank string)
    * `''`(just '' string, not empty)
    * `'xxx'`
    * 123
    * ...
   
Clearly distinguish them use one `Default` column in UI may be difficult.

"About `DROP DEFAULT`, well, how many people do that, really ?"
 -->
`DROP DEFAULT` then `DEFAULT not exists`, `SACE` UI will show blank.

**Comment by jakubboucek** [comment]
Currently is unable to add row without `DEFAULT VALUE`, because when I leave `DEFAULT` column empty, it cause to invalid SQL command.

![image](https://user-images.githubusercontent.com/1657322/159588238-fb0d4ae8-e10f-4e21-8d6b-ce15d313b625.png)

When I leave `DEFAULT` field empty, it MUST not to generate `DEFAULT` definition to ALTER SQL command.

First changes from #1163 work great, but now it has wrong behavior.

**Comment by Lichmaker** [comment]
> Currently is unable to add row without `DEFAULT VALUE`, because when I leave `DEFAULT` column empty, it cause to invalid SQL command.
> 
> ![image](https://user-images.githubusercontent.com/1657322/159588238-fb0d4ae8-e10f-4e21-8d6b-ce15d313b625.png)
> 
> When I leave `DEFAULT` field empty, it MUST not to generate `DEFAULT` definition to ALTER SQL command.
> 
> First changes from #1163 work great, but now it has wrong behavior.

I have the same issues  :(
AppStore  Version 3.5.1  Build 20031
Plz fix this.

**Comment by Jason-Morcos** [comment]
> @Jason-Morcos Anyway, I think the display in the 'Structure' tab needs to be clarified :
> 
> * Adding '' in case of `default ''` is set.
> * Globally, for any `default 'xxxx'`, add the '' if required:
>   
>   * `DEFAULT 'SACE'`  -->  `'SACE'`   (With '')
>   * `DEFAULT 123`  -->  `123`    (Without '')
> * Adding NULL (grey?) if NULL is not allowed and "DEFAULT VALUE" is not set?
> 
> About `DROP DEFAULT, well, how many people do that, really ?

I think adding the quotes like you suggested is a great idea. I think if no default is set, the default display should say `None` just like we do for the Extra column. That would be consistent with the rest of the UI, imo. Those two pieces together should provide a lot of clarity to something that is objectively confusing. I would treat a `DEFAULT NULL` as showing `NULL` in the default display, but I wouldn't show `NULL` in cases when no default is set - let's try to _exactly_ display what the user actually chose when creating the table - MySQLs inferred defaults should show `None`, because no default value was specified _by the user_. I think that this way we're at least consistent with MySQL, even if MySQL itself may be a little inconsistent.

**Comment by Jason-Morcos** [comment]
As to those referencing the actual bug present in 3.5.1 with default value editing, that should be fixed in the latest beta of 3.5.2 and will be fixed in the upcoming release. This issue here is purely a display one afaik - the display is unclear.
